common/lib/dependabot/clients/azure.rb

```diff
@@ -253,8 +255,9 @@ def compare(previous_tag, new_tag, type)
         JSON.parse(response.body).fetch("value")
       end
 
+      sig { params(url: String).returns(Excon::Response) }
       def get(url)
-        response = nil
+        response = T.let(nil, T.nilable(Excon::Response))
```

Comment:
Are these `T.let`s and `T.must`s checked at runtime as well, or are these also just compile-time checks?

Reply:
`T.let` and `T.must` are both static and runtime checks. Unfortunately, it's not possible to satisfy Sorbet's type checks with just static checks in this specific instance.

Sorbet infers `response = nil` is of type `nil`, and then complains that `Excon::Result` cannot be assigned to type `nil`. So `response` has to be defined as `T.nilable(Excon::Response)`.

Later we make calls to `response.status` and `response.body`. I made the `response.status` calls use the safe navigation operator `&.`. But before returning from the method, I assert that `response` is not `nil` by calling `T.must(response)`. The alternative is propagating the nil all the way up the stack, which will require a lot of condditionals where the result of a method call _might_ be `nil`.

Reply:
I see, thanks for explaining! It might make sense to enable the runtime checking only in tests and not in prod yet if there's a way to make that configurable? That way we reduce the chance of introducing any new issues, and also I believe there's a (small?) cost incurred when enabling runtime checks?

Reply:
Yeah, I can add that configuration somewhere in the root of `common`. There is a small performance penalty for runtime type checks, but it's somewhere in the region of 5%[^1]

> we have runtime type checking turned on even for our most performance-sensitive request paths, and the slowdown from runtime type checking amounts to about 5%.

And if we ever end up using the Sorbet compiler to AOT compile `dependabot-core` we'll see huge performance _increases_ with runtime type checks[^2]

[^1]: https://blog.jez.io/runtime-type-checking/
[^2]: https://sorbet.org/blog/2021/07/30/open-sourcing-sorbet-compiler